### PR TITLE
Fix python3 issues.

### LINKED ISSUE #239: Python3 issues
While trying to use ducktape with python3, we encountered some issues.

I've fixed them in #238 `BackgroundThreadService` and `RemoteAccount#signal`

### PR DESCRIPTION
**Pull request by ivandasch**
Fix #239

### CODE REVIEW

**Comment by ivandasch** [comment]
[clabot:check]

**Comment by ivandasch** [comment]
[clabot:check]

File: systests/cluster/test_remote_account.py [hunk 1]
```diff
@@ -509,6 +511,29 @@ def test_monitor_log_exception(self):
...
+
+        def get_pids():
+            pid_cmd = "ps ax | grep -i nc | grep -v grep | awk '{print $1}'"
+
+            return list(node.account.ssh_capture(pid_cmd, callback=int))
+
+        node = self.account_service.nodes[0]
+
+        # Run TCP service using netcat
+        node.account.ssh_capture("nohup nc -l -p 5000 > /dev/null 2>&1 &")
```
Comment (andrewegel) [inline systests/cluster/test_remote_account.py:526]:
```suggestion
        node.account.ssh_capture("nohup nc -l -p 5000 > /dev/null 2>&1 &; disown")
```

I've encountered a behavior change on EL8 Platforms (CentOS-8 and RHEL8) where the SSHD daemon may terminate the shell, but the SSHD "session" for that connection remains open for the remainder of

**Comment by ivandasch** [comment]
@stan-confluent @andrewegel Hi, guys, thanks for a review. 
@ewencp Could you please take a look on this? Is there any objection to merge this patch?

**Pull request merged**